### PR TITLE
Add auth header to get requests

### PR DESCRIPTION
**Pull request by Drapegnik**
For supporting private repos

### CODE REVIEW

**Comment by benedfit** [comment]
@bwyap any chance this could get reviewed and merged?

**Comment by benyap** [comment]
@benedfit will look into it!

File: src/api.ts [hunk 1]
```diff
@@ -3,6 +3,11 @@ import axios from "axios";
 import { DeploymentStatus, Deployment, DeploymentState } from "./types";
 import { githubAPI } from "./constants";
 
+const getDefaultHeaders = (token: string) => ({
+  Authorization: `token ${token}`,
```
Comment (benyap) [inline src/api.ts:7]: There should be an option to **not** have the auth header if we want to access public repos. Seems like currently an auth header will always be sent with the request.

File: src/api.ts [hunk 2]
```diff
@@ -15,24 +20,32 @@ import { githubAPI } from "./constants";
  */
 export const getDeployments = ({
   user,
-  repo
+  repo,
+  token
 }: {
   user: string;
   repo: string;
+  token: string;
```
Comment (benyap) [inline src/api.ts]: Might be good to add `token` to the `Arguments` interface in `types.d.ts` so the definition is more accurate.

**Review by benyap — CHANGES_REQUESTED** [review]
Hi @Drapegnik, thanks for your contribution. Apologies for not getting back, didn't get a chance to review this earlier and it got lost in my list of things to do 😅Just a few small changes required I reckon and it should be good to go!

EDIT: Would be good if you could document the new API in the README too.

File: src/api.ts (hunk 1 above)
Comment (benedfit) [reply src/api.ts:7]: Would it do any harm to always pass it?

Comment (Drapegnik) [reply src/api.ts:7]: @benedfit, of course no.
@bwyap, I've tested it with my public repo. we can always pass auth token

File: src/api.ts (hunk 2 above)
Comment (Drapegnik) [reply src/api.ts]: it's already there

File: src/api.ts (hunk 1 above)
Comment (Drapegnik) [reply src/api.ts:7]: > `--token <token>` - This argument is required.

and valid token is already requirement for all repos

**Pull request merged**